s3_management/manage.py

```diff
@@ -348,10 +351,13 @@ def from_S3(cls: Type[S3IndexType], prefix: str) -> S3IndexType:
                 for pattern in ACCEPTED_SUBDIR_PATTERNS
             ]) and obj.key.endswith(ACCEPTED_FILE_EXTENSIONS)
             if is_acceptable:
+                response = obj.meta.client.head_object(Bucket=BUCKET.name, Key=obj.key, ChecksumMode="ENABLED")
+                sha256 = response.get("ChecksumSHA256")
```

Comment:
Probably worth a comment explaining why we go through the hassle here.

Something like:
`# Add PEP 503-compatible hashes to URLs to allow clients to avoid re-downloads, if possible.